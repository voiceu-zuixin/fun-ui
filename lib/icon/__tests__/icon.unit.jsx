import 'jsdom-global/register'
import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '../icon'
import {mount} from 'enzyme'


describe('icon', () => {
    it('render successfully', () => {
        const json = renderer.create(<Icon name='alipay' />).toJSON()
        // 期望json符合快照
        expect(json).toMatchSnapshot()
    })
    it('onClick',()=>{
        const fn = jest.fn()
        // let n = 1
        // const fn = ()=>{
        //     n=2
        // }
        const component = mount(<Icon name='alipay' onClick={fn}/>)
        component.find('svg').simulate('click')
        // expect(n).toEqual(2)
        expect(fn).toBeCalled()
    })
})