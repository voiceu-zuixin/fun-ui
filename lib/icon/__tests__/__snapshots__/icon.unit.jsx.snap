// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render successfully 1`] = `
<svg
  className="voice-ui-icon"
>
  npm i jsdom --save-dev --sass_binary_site=https://npm.taobao.org/mirrors/jsdom/
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
